### ISSUE TITLE
Tests to Check Agents Meet Certain Instructions  

### ISSUE THREAD

**Issue by volodya-lombrozo** [objectionary/opeo-maven-plugin#381]
It would be great to add semi-integration tests that would check that the new agents works as expected. Right now we have just a set of unit tests or full integration tests, but we could add something like the following:
```yaml
opcodes:
  - ILOAD 423
  - ILOAD 7
  - IADD
agents:
  - AdditionAgent
  - SomeOtherAgent
eo:
  432.add 7
```
This tests will help us understand how agents work, by illustrative examples.

**Note!** In the opcodes section we may want to have EO objects too, not only `opcodes`. Maybe it should be `input` section, instead of `opcodes`.

You can read about the idea right [here](https://github.com/objectionary/opeo-maven-plugin/pull/379#issuecomment-2285686327).

**Comment by 0pdd** [comment]
@volodya-lombrozo 2 puzzles [#403](https://github.com/objectionary/opeo-maven-plugin/issues/403), [#404](https://github.com/objectionary/opeo-maven-plugin/issues/404) are still not solved.

**Comment by 0pdd** [comment]
@volodya-lombrozo 3 puzzles [#403](https://github.com/objectionary/opeo-maven-plugin/issues/403), [#407](https://github.com/objectionary/opeo-maven-plugin/issues/407), [#408](https://github.com/objectionary/opeo-maven-plugin/issues/408) are still not solved; solved: [#404](https://github.com/objectionary/opeo-maven-plugin/issues/404).
